### ISSUE TITLE
 Create an Opcodes class

### ISSUE THREAD

**Issue by alex0112** [alex0112/abacus#6]
# Opcode Abstraction

## High Level Description

Create a little object (there's an initial draft in `src/opcodes.py`) that provides a convenient interface for an opcode. (Using the word "opcode" in this context to mean an instruction or piece of memory like `+1020` or `-2099` etc)

### Example
This task would be complete when another programmer can import the module create/use the object like:

```python
opcode = Opcode("+1042")

opcode.operation ## returns `10` or `READ` or something. Figure it out
opcode.sign ## returns `+` or `-`
opcode.operand ## returns `42` 

bad_opcode = Opcode("9999999") ## This should throw an exception of some kind 
```

### Potential Sub-Tasks

- [x] Write a unit test for each of these cases
- [x] Read a raw string like "+1042" and turn it into an Opcode object
- [x] Create an attribute for the sign of the opcode
- [x] Create an attribute for the operator (The instruction like `READ` which is encoded as `10`)
   - We could maybe consider having a human friendly version like `READ` and a machine friendly version like `10`
- [x] Create an attribute for the operand (the second part of the opcode)
- [x] Handle cases where the opcode is *negative* *positive* *unsigned* and *invalid*

**Comment by jordanpaxman** [comment]
I can start working on this. However I am still getting used to github, should I start working on a separate file called opcode.py on the shared codespace? Or where do I work on this? @alex0112

**Comment by jordanpaxman** [comment]
I am reading the instructions more thoroughly and I get it now. I see the src/opcode.py file. So how do we want to delegate these tasks out to the team?

**Comment by alex0112** [comment]
@jordanpaxman thanks for taking a look. I met with Jackson today and we discussed the overall design of the project. We're going to all have a meeting tonight at 6:30pm where we can go over everything and assign tasks, but I'll make a note that you called dibs on this one if you still want it by then

I'll be making another few tasks as github issues and we can organize and divide labor up from there, that's the plan for the meeting tonight.

**Comment by alex0112** [comment]
@jordanpaxman as far as how to work on this, the typical way is to create a git branch off of the main branch (wait until my current code is merged tho) and then create your own branch to work on. Once your work is done on that branch you can create a pull request which the team can review and once it's approved (however we want to do that, if at all) then we'll merge it to the main branch and it will be part of the program.

There are two ways to do this. If you're working on a project where you're not a member: https://www.youtube.com/watch?v=jRLGobWwA3Y (this involves forking the project)

However since we all own this project you can just create a new branch on the project directly. I couldn't find a video on it right this second but feel free to google around. The basic workflow is:
- clone the project to your local machine
- make a new branch (`git checkout -b yourbranchname`) off of main (it's common to name something like it `feature/whatever-feature-you-are-adding` or `your-name/new-feature-name`)
- make your changes in that branch
- run `git push` (that will complain that there's no branch on github with your branch name, so you need to run the command it suggests to create it on github)
- Once that's done go to github on the web and make a new pull request from the branch you just pushed. This will let the rest of the team review it
